src/ClipperMom.sol

```
@@ -72,9 +110,51 @@ contract ClipperMom {
         ClipLike(clip).file("stopped", level);
+        // If governance changes the status of the breaker we want to lock for one hour
+        // the permissionless function so the osm can pull new nxt price to compare
+        locked[ClipLike(clip).ilk()] = block.timestamp + 1 hours;
         emit SetBreaker(clip, level);
     }
+
+    /**
+        The following implements a permissionless circuit breaker in case the price reported by an oracle
+        for a particular collateral type has dropped more than a governance-defined % from 1 hour to the next.
+
+        SetPriceDropTolerance sets a % (a RAY number between 0 and 1) for a specific collateral type
+        It then gets the price of the ilk from the spotter and caches that price, the current time, and
+        the % tolerance.
+            
+
+        tripBreaker takes the address of the ilk's clipper as well as the ilk identifier.
+        it then gets the current and next price and checks:
+          - the next price < current price
+          - the currentPrice * (tolerance % * 100) is less than the currentPrice - next price
+            - i.e., the acceptable drop in price < the actual drop
+          - If the drop is unacceptable, it stops auctions for the current ilk and allows a later retry
+        
+          - Edge cases: 
+            - The clipper is for a different ilk than the ilk whose price we are breaking -> require the clipper's ilk == ilk
+    
+    */
+    function tripBreaker(address clip) external {
+        ClipLike clipper = ClipLike(clip);
+        bytes32 ilk = clipper.ilk();
+        require(tolerance[ilk] > 0, "ClipperMom/invalid-ilk-break");
+        require(block.timestamp > locked[ilk], "ClipperMom/temporary-locked");
+      
+        (uint256 cur, uint256 nxt) = getPrices(ilk);
+
+        require(nxt < rmul(cur, sub(RAY, tolerance[ilk])), "ClipperMom/price-within-bounds");
+        clipper.file("stopped", 2);
+        emit SetBreaker(clip, 2);
```

This does not appear to set the `locked` value, and further attempts to set this when the price is within bounds causes a `revert`. Is this intended behavior?

---

Not sure if I'm following this. If the price is within bounds this function shouldn't pass.
If it was already called and the price is out of the boundaries it should pass but will be setting up the same stopped to 2 multiple times which shouldn't be a problem.

---

I'm with Gonzalo here. We only care about setting `locked` if governance wants to override the circuit breaker. I also don't see a problem with reverting whenever the trip condition isn't met...the alternative is to silently return without making any change...I'd think a caller would appreciate the quick feedback of a revert if they couldn't trip the breaker like they expected to.

---

Yeah I guess I was thinking that there's no recovery from this without governance intervention. In a legitimate downturn that can put us into a very bad situation at a time when the protocol is already under extreme duress.

I'm happy being the minority opposition to this type of breaker mechanism, but I want to express my disfavor for it.

---

Keep in mind that Oracles is already considering reducing the number of `poke`'s on certain collateral types as a cost-saving measure, so each update won't necessarily be an hour apart and price updates may be wider than we'd expect in just the course of an hour. We can offload these calculations to risk, of course, but we should be very clear about it in the handoff.

---

I don't understand what Brian is opposed to. Are you suggesting we should have a way to untrip (I hope I said it right) the breaker without governance intervention? I thought that was a settled matter.

---

Yes. For the record. I've never been in favor of an unattended lockdown. I prefer a solution that slows down the damage and gives governance time to react in the case of an emergency to a solution that can lock up the system and demand a governance action to resume normal operation. 

I am under the impression that this is temporary until we can solve the oracle attack issue with a more robust solution.

---

> Keep in mind that Oracles is already considering reducing the number of `poke`'s on certain collateral types as a cost-saving measure

I think this is only for `median`s and we are still thinking on poking each `osm` once an hour.



> I am under the impression that this is temporary until we can solve the oracle attack issue with a more robust solution.

Yes this will be a v1 until we can get a module that can disable and enable auctions permissionless.

---

Well regarding the last one, I will better say we will research if there is a good way for enabling auctions back in a permissionless way.

---

I agree with Brian that this is not ideal and needs to be changed soon.

---

We will see if there is an automation function that can actually re-enable auctions in a safe way. Hopefully we find the way to do it.